app/pods/components/player-code-challenge/component.js

```diff
@@ -116,14 +117,40 @@ export default class CodeChallengeComponent extends Component {
   @restartableTask
   *runCodeTask (config) {
     this.set('api.headers.hackJwt', this.get('currentUser.user.hackJwt'))
-    return yield this.get("api").request("code_challenges/submit", {
+    const payload = yield this.get("api").request("code_challenges/submit", {
       method: "POST",
       data: {
         custom_input: config.input,
         source: config.source,
         language: config.lang,
       },
     });
+
+    const submissionId = payload.submissionId
+    return yield this.fetchSubmissionStatusTask(submissionId)
+  }
+  
+  /**
+   * Fetch the submission status
+   * 
+   * @param {*} submissionId 
+   * @returns {Promise} API result Promise
+   */
+
+  async fetchSubmissionStatusTask (submissionId) {
```

Comment:
This isn't a ember-concurreny `task`, so let's now prefix it with task. The API is different.

Reply:
^ fixed typo

Reply:
Removed this method altogether